### PR TITLE
builtin function for parsing links from html

### PR DESCRIPTION
**Pull request by sfultong**

### CODE REVIEW

**Comment by pchiusano** [comment]
Can you fix Travis error? Looks like just need to update cabal file.

File: node/src/Unison/Runtime/ExtraBuiltins.hs [hunk 1]
```diff
@@ -81,5 +98,29 @@ makeAPI blockStore crypto = do
...
        in (r, Just (I.Primop 3 op), unsafeParseType "String -> String -> Store String String -> Remote Unit", prefix "insert")
+     , let r = R.Builtin "Html.getLinks"
+           op [html] = do
+             html' <- whnf html
+             pure $ case html' of
+               Term.Text' h -> Term.vector' . Vector.fromList . map linkToTerm
+                 $ Html.getLinks h
+               x -> Term.ref r `Term.app` x
+           op _ = fail "Html.getLinks unpossible"
+       in (r, Just (I.Primop 1 op), unsafeParseType "String -> Vector Link", prefix "getLinks")
```
Comment (pchiusano) [inline node/src/Unison/Runtime/ExtraBuiltins.hs]: Should be Text not String in the type. String is not a Unison type. If parser is parsing String as Text that should be updated.

Comment (pchiusano) [inline node/src/Unison/Runtime/ExtraBuiltins.hs]: Can you add a test that evaluates the function via Unison? That will make sure the type declaration and your binding to the underlying fn is correct.

File: node/src/Unison/Runtime/Html.hs [hunk 2]
```diff
@@ -0,0 +1,26 @@
+module Unison.Runtime.Html where
+
+import Data.Maybe (listToMaybe, catMaybes, mapMaybe)
+import Data.Text (Text, toLower, pack)
+import Text.HTML.TagSoup
```
Comment (pchiusano) [inline node/src/Unison/Runtime/Html.hs]: Can you explicitly import any operators you are using and/or import the module qualified? Most readers of this code are unlikely to be familiar with what tag soup provides

Comment (pchiusano) [inline node/src/Unison/Runtime/Html.hs]: Reminder to myself that I should create a style guide.

**Comment by pchiusano** [comment]
Lgtm, made a couple minor comments.

File: node/src/Unison/Runtime/Html.hs (hunk 2 above)
Comment (sfultong) [inline node/src/Unison/Runtime/Html.hs]: Yeah, a style guide would be good. Is there already a Haskell style guide you roughly base your style on?

Comment (pchiusano) [inline node/src/Unison/Runtime/Html.hs]: Not that I know of.

**Comment by sfultong** [comment]
@pchiusano I'm ready for another review

File: node/src/Unison/Runtime/JournaledSet.hs [hunk 3]
```diff
@@ -0,0 +1,72 @@
+{-# Language DeriveGeneric #-}
```
Comment (pchiusano) [inline node/src/Unison/Runtime/JournaledSet.hs]: If this module isn't being used anywhere, just delete it. Also, for journaled data structures that we do have in the codebase, I want to move away from having the convenience methods, in favor of just using `Journal.update/updateAsync/updateNowFlushAsync` directly.

**Comment by pchiusano** [comment]
👍

**Pull request merged**